include/openthread/link.h

```diff
@@ -379,18 +379,19 @@ typedef struct otMacCounters
  */
 typedef struct otActiveScanResult
 {
-    otExtAddress    mExtAddress;    ///< IEEE 802.15.4 Extended Address
-    otNetworkName   mNetworkName;   ///< Thread Network Name
-    otExtendedPanId mExtendedPanId; ///< Thread Extended PAN ID
-    otSteeringData  mSteeringData;  ///< Steering Data
-    uint16_t        mPanId;         ///< IEEE 802.15.4 PAN ID
-    uint16_t        mJoinerUdpPort; ///< Joiner UDP Port
-    uint8_t         mChannel;       ///< IEEE 802.15.4 Channel
-    int8_t          mRssi;          ///< RSSI (dBm)
-    uint8_t         mLqi;           ///< LQI
-    unsigned int    mVersion : 4;   ///< Version
-    bool            mIsNative : 1;  ///< Native Commissioner flag
-    bool            mDiscover : 1;  ///< Result from MLE Discovery
+    otExtAddress    mExtAddress;     ///< IEEE 802.15.4 Extended Address
+    otNetworkName   mNetworkName;    ///< Thread Network Name
+    otExtendedPanId mExtendedPanId;  ///< Thread Extended PAN ID
+    otSteeringData  mSteeringData;   ///< Steering Data
+    uint16_t        mPanId;          ///< IEEE 802.15.4 PAN ID
+    uint16_t        mJoinerUdpPort;  ///< Joiner UDP Port
+    uint8_t         mChannel;        ///< IEEE 802.15.4 Channel
+    int8_t          mRssi;           ///< RSSI (dBm)
+    uint8_t         mLqi;            ///< LQI
+    unsigned int    mVersion : 4;    ///< Version
+    bool            mIsNative : 1;   ///< Native Commissioner flag
+    bool            mDiscover : 1;   ///< Result from MLE Discovery
+    bool            mIsJoinable : 1; ///< Joining Permitted flag
```

Comment:
Should `mIsJoinable` be wrapped with a check to see if `OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE` is enabled?  Is see that it was added as part of this PR but the name does not really imply that it is beacon specific.

Reply:
@romacdon I left it out intentionally, because this structure still has other elements not used in a active scan result. I wanted to avoid fragmentation of this structure.

Reply:
Just a note on this: we cannot add or use `#if OPENTRHEAD_CONFIG...` ` checks in any of public OT headers. We can document that certain behavior require certain configs (some more info in https://github.com/openthread/openthread/issues/4333).